File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    
    if list1 == nil {
        return list2
    } else if list2 == nil {
        return list1
    }
    
    head := &ListNode{}
    currentNode := head
    
    for (list1 != nil) && (list2 != nil) {
        if list1.Val < list2.Val {
            currentNode.Next = list1
            list1 = list1.Next
            currentNode = currentNode.Next
        } else {
            currentNode.Next = list2
            list2 = list2.Next
            currentNode = currentNode.Next
        }
    }
    
    for list1 != nil {
        currentNode.Next = list1
        list1 = list1.Next
        currentNode = currentNode.Next
    }
    
    for list2 != nil {
        currentNode.Next = list2
        list2 = list2.Next
        currentNode = currentNode.Next
    }
    
    return head.Next
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode result = new ListNode(0);
        ListNode nextPtr = result;
        while(list1 != null && list2 != null) {
            if(list1.val < list2.val) {
                nextPtr.next = list1;
                nextPtr = nextPtr.next;
                list1 = list1.next;
            } else {
                nextPtr.next = list2;
                nextPtr = nextPtr.next;
                list2 = list2.next;
            }
        }

        if(list1 != null) {
            nextPtr.next = list1;
        } else if(list2 != null) {
            nextPtr.next = list2;
        }

        return result.next;
    }
}